src/azure-cli/azure/cli/command_modules/acr/agentpool.py

```diff
@@ -84,9 +88,9 @@ def acr_agentpool_delete(cmd,
     user_confirmation("Are you sure you want to delete the agentpool '{}' in registry '{}'?".format(
         agent_pool_name, registry_name), yes)
     try:
-        response = client.delete(resource_group_name=resource_group_name,
-                                 registry_name=registry_name,
-                                 agent_pool_name=agent_pool_name)
+        response = client.begin_delete(resource_group_name=resource_group_name,
```

Comment:
The result is not polled.

Then at L104, `delete_agentpool_with_polling` creates another `LROPoller`. This causes `Message: The subscription '00000000-0000-0000-0000-000000000000' could not be found.` error: 

https://dev.azure.com/azclitools/public/_build/results?buildId=199090&view=logs&jobId=76ded334-3544-5630-8508-787a6cf96d29&j=76ded334-3544-5630-8508-787a6cf96d29&t=c62dd615-c6c6-5fdb-a846-07cb3711ef74

```
2024-10-22T02:44:50.2849766Z _________________ AcrAgentPoolCommandsTests.test_acr_agentpool _________________
2024-10-22T02:44:50.2850177Z [gw0] linux -- Python 3.11.8 /opt/az/bin/python3
2024-10-22T02:44:50.2850492Z self = <azure.cli.testsdk.base.ExecutionResult object at 0x7f3290732150>
2024-10-22T02:44:50.2850820Z cli_ctx = <azure.cli.core.mock.DummyCli object at 0x7f3292f7a610>
2024-10-22T02:44:50.2851257Z command = 'acr agentpool delete -n agents2 -r clireg000002 -y'
2024-10-22T02:44:50.2851538Z expect_failure = False
2024-10-22T02:44:50.2851646Z 
2024-10-22T02:44:50.2851910Z     def _in_process_execute(self, cli_ctx, command, expect_failure=False):
2024-10-22T02:44:50.2852202Z         from io import StringIO
2024-10-22T02:44:50.2852491Z         from vcr.errors import CannotOverwriteExistingCassetteException
2024-10-22T02:44:50.2852748Z     
2024-10-22T02:44:50.2853053Z         if command.startswith('az '):
2024-10-22T02:44:50.2853300Z             command = command[3:]
2024-10-22T02:44:50.2853508Z     
2024-10-22T02:44:50.2853723Z         stdout_buf = StringIO()
2024-10-22T02:44:50.2853965Z         logging_buf = StringIO()
2024-10-22T02:44:50.2854190Z         try:
2024-10-22T02:44:50.2854472Z             # issue: stderr cannot be redirect in this form, as a result some failure information
2024-10-22T02:44:50.2854774Z             # is lost when command fails.
2024-10-22T02:44:50.2855091Z >           self.exit_code = cli_ctx.invoke(shlex.split(command), out_file=stdout_buf) or 0
2024-10-22T02:44:50.2855250Z 
2024-10-22T02:44:50.2855640Z /opt/az/lib/python3.11/site-packages/azure/cli/testsdk/base.py:302: 
2024-10-22T02:44:50.2855962Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2024-10-22T02:44:50.2856368Z /opt/az/lib/python3.11/site-packages/knack/cli.py:245: in invoke
2024-10-22T02:44:50.2856673Z     exit_code = self.exception_handler(ex)
2024-10-22T02:44:50.2857120Z /opt/az/lib/python3.11/site-packages/azure/cli/core/__init__.py:129: in exception_handler
2024-10-22T02:44:50.2857436Z     return handle_exception(ex)
2024-10-22T02:44:50.2857710Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2024-10-22T02:44:50.2857832Z 
2024-10-22T02:44:50.2858523Z ex = ResourceNotFoundError("(SubscriptionNotFound) The subscription '00000000-0000-0000-0000-000000000000' could not be found.\nCode: SubscriptionNotFound\nMessage: The subscription '00000000-0000-0000-0000-000000000000' could not be found.")
2024-10-22T02:44:50.2858968Z args = (), kwargs = {}
2024-10-22T02:44:50.2859061Z 
2024-10-22T02:44:50.2859465Z     def _handle_main_exception(ex, *args, **kwargs):  # pylint: disable=unused-argument
2024-10-22T02:44:50.2859815Z         if isinstance(ex, CannotOverwriteExistingCassetteException):
2024-10-22T02:44:50.2860154Z             # This exception usually caused by a no match HTTP request. This is a product error
2024-10-22T02:44:50.2860476Z             # that is caused by change of SDK invocation.
2024-10-22T02:44:50.2860737Z             raise ex
2024-10-22T02:44:50.2860936Z     
2024-10-22T02:44:50.2861159Z >       raise CliExecutionError(ex)
2024-10-22T02:44:50.2861514Z E       azure.cli.testsdk.exceptions.CliExecutionError: The CLI throws exception ResourceNotFoundError during execution and fails the command.
2024-10-22T02:44:50.2861723Z 
2024-10-22T02:44:50.2862130Z /opt/az/lib/python3.11/site-packages/azure/cli/testsdk/patches.py:35: CliExecutionError
2024-10-22T02:44:50.2862297Z 
2024-10-22T02:44:50.2862562Z During handling of the above exception, another exception occurred:
2024-10-22T02:44:50.2862699Z 
2024-10-22T02:44:50.2863019Z self = <command_modules.acr.tests.latest.test_acr_agentpool_commands.AcrAgentPoolCommandsTests testMethod=test_acr_agentpool>
2024-10-22T02:44:50.2863666Z resource_group = 'clitest.rg000001'
2024-10-22T02:44:50.2863776Z 
2024-10-22T02:44:50.2864004Z     @ResourceGroupPreparer()
2024-10-22T02:44:50.2864265Z     def test_acr_agentpool(self, resource_group):
2024-10-22T02:44:50.2864550Z         # Agentpool prerequisites for agentpool testing
2024-10-22T02:44:50.2864807Z         self.kwargs.update({
2024-10-22T02:44:50.2865188Z             'registry_name': self.create_random_name('clireg', 20),
2024-10-22T02:44:50.2865546Z             'agents1_name': 'agents1',
2024-10-22T02:44:50.2865880Z             'agents2_name': 'agents2',
2024-10-22T02:44:50.2866196Z             'rg_loc': 'eastus',
2024-10-22T02:44:50.2866500Z             'sku': 'Premium',
2024-10-22T02:44:50.2866817Z             'vnet_name': 'agentvnets',
2024-10-22T02:44:50.2867156Z             'subnet_name': 'agentsubnets'
2024-10-22T02:44:50.2867386Z         })
2024-10-22T02:44:50.2867784Z         self.cmd('acr create -n {registry_name} -g {rg} -l {rg_loc} --sku {sku}',
2024-10-22T02:44:50.2868219Z                  checks=[self.check('name', '{registry_name}'),
2024-10-22T02:44:50.2868611Z                          self.check('location', '{rg_loc}'),
2024-10-22T02:44:50.2869000Z                          self.check('adminUserEnabled', False),
2024-10-22T02:44:50.2869407Z                          self.check('sku.name', '{sku}'),
2024-10-22T02:44:50.2869787Z                          self.check('sku.tier', '{sku}'),
2024-10-22T02:44:50.2870203Z                          self.check('provisioningState', 'Succeeded')])
2024-10-22T02:44:50.2870723Z         response = self.cmd('network vnet create -n {vnet_name} --subnet-name {subnet_name} -g {rg} -l {rg_loc}',
2024-10-22T02:44:50.2871212Z                             checks=[self.check('newVNet.name', '{vnet_name}'),
2024-10-22T02:44:50.2871656Z                                     self.check('newVNet.location', '{rg_loc}'),
2024-10-22T02:44:50.2872445Z                                     self.check('newVNet.provisioningState', 'Succeeded'),
2024-10-22T02:44:50.2872965Z                                     self.check('newVNet.subnets[0].name', '{subnet_name}'),
2024-10-22T02:44:50.2873504Z                                     self.check('newVNet.subnets[0].provisioningState', 'Succeeded')]).get_output_in_json()
2024-10-22T02:44:50.2873841Z         self.kwargs.update({
2024-10-22T02:44:50.2874239Z             'subnet_id': response['newVNet']['subnets'][0]['id']
2024-10-22T02:44:50.2874501Z         })
2024-10-22T02:44:50.2874694Z     
2024-10-22T02:44:50.2874938Z         # Create a default S1 count 1 agentpool.
2024-10-22T02:44:50.2875377Z         self.cmd('acr agentpool create -n {agents1_name} -r {registry_name}',
2024-10-22T02:44:50.2875796Z                  checks=[self.check('name', '{agents1_name}'),
2024-10-22T02:44:50.2876164Z                          self.check('count', 1),
2024-10-22T02:44:50.2876516Z                          self.check('tier', 'S1'),
2024-10-22T02:44:50.2876931Z                          self.check('virtualNetworkSubnetResourceId', 'None'),
2024-10-22T02:44:50.2877359Z                          self.check('provisioningState', 'Succeeded'),
2024-10-22T02:44:50.2877739Z                          self.check('os', 'Linux'),
2024-10-22T02:44:50.2878105Z                          self.check('location', '{rg_loc}'),
2024-10-22T02:44:50.2878468Z                          self.check('resourceGroup', '{rg}')])
2024-10-22T02:44:50.2878719Z     
2024-10-22T02:44:50.2878964Z         # Create a S2 tier agentpool in a VNET.
2024-10-22T02:44:50.2879465Z         self.cmd('acr agentpool create -n {agents2_name} -r {registry_name} --tier s2 --subnet-id {subnet_id}',
2024-10-22T02:44:50.2879941Z                  checks=[self.check('name', '{agents2_name}'),
2024-10-22T02:44:50.2880311Z                          self.check('count', 1),
2024-10-22T02:44:50.2880670Z                          self.check('tier', 'S2'),
2024-10-22T02:44:50.2881291Z                          self.check('virtualNetworkSubnetResourceId', '{subnet_id}'),
2024-10-22T02:44:50.2881868Z                          self.check('provisioningState', 'Succeeded'),
2024-10-22T02:44:50.2882251Z                          self.check('os', 'Linux'),
2024-10-22T02:44:50.2882625Z                          self.check('location', '{rg_loc}'),
2024-10-22T02:44:50.2882989Z                          self.check('resourceGroup', '{rg}')])
2024-10-22T02:44:50.2883221Z         # List agentpools
2024-10-22T02:44:50.2883555Z         self.cmd('acr agentpool list -r {registry_name}',
2024-10-22T02:44:50.2883920Z                  checks=[self.check('[0].name', '{agents2_name}'),
2024-10-22T02:44:50.2884283Z                          self.check('[1].name', '{agents1_name}')])
2024-10-22T02:44:50.2884508Z     
2024-10-22T02:44:50.2884699Z         # Delete Agent 2
2024-10-22T02:44:50.2885061Z >       self.cmd('acr agentpool delete -n {agents2_name} -r {registry_name} -y')
2024-10-22T02:44:50.2885201Z 
2024-10-22T02:44:50.2885638Z /opt/az/lib/python3.11/site-packages/azure/cli/command_modules/acr/tests/latest/test_acr_agentpool_commands.py:67: 
2024-10-22T02:44:50.2885982Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2024-10-22T02:44:50.2886431Z /opt/az/lib/python3.11/site-packages/azure/cli/testsdk/base.py:176: in cmd
2024-10-22T02:44:50.2886808Z     return execute(self.cli_ctx, command, expect_failure=expect_failure).assert_with_checks(checks)
2024-10-22T02:44:50.2887346Z /opt/az/lib/python3.11/site-packages/azure/cli/testsdk/base.py:251: in __init__
2024-10-22T02:44:50.2887755Z     self._in_process_execute(cli_ctx, command, expect_failure=expect_failure)
2024-10-22T02:44:50.2888270Z /opt/az/lib/python3.11/site-packages/azure/cli/testsdk/base.py:314: in _in_process_execute
2024-10-22T02:44:50.2888603Z     raise ex.exception
2024-10-22T02:44:50.2888997Z /opt/az/lib/python3.11/site-packages/knack/cli.py:233: in invoke
2024-10-22T02:44:50.2889314Z     cmd_result = self.invocation.execute(args)
2024-10-22T02:44:50.2889792Z /opt/az/lib/python3.11/site-packages/azure/cli/core/commands/__init__.py:666: in execute
2024-10-22T02:44:50.2890109Z     raise ex
2024-10-22T02:44:50.2890550Z /opt/az/lib/python3.11/site-packages/azure/cli/core/commands/__init__.py:733: in _run_jobs_serially
2024-10-22T02:44:50.2890941Z     results.append(self._run_job(expanded_arg, cmd_copy))
2024-10-22T02:44:50.2891463Z /opt/az/lib/python3.11/site-packages/azure/cli/core/commands/__init__.py:703: in _run_job
2024-10-22T02:44:50.2891806Z     result = cmd_copy(params)
2024-10-22T02:44:50.2892295Z /opt/az/lib/python3.11/site-packages/azure/cli/core/commands/__init__.py:336: in __call__
2024-10-22T02:44:50.2892647Z     return self.handler(*args, **kwargs)
2024-10-22T02:44:50.2893148Z /opt/az/lib/python3.11/site-packages/azure/cli/core/commands/command_operation.py:121: in handler
2024-10-22T02:44:50.2893511Z     return op(**command_args)
2024-10-22T02:44:50.2894096Z /opt/az/lib/python3.11/site-packages/azure/cli/command_modules/acr/agentpool.py:104: in acr_agentpool_delete
2024-10-22T02:44:50.2894561Z     return delete_agentpool_with_polling(cmd, client, agent_pool_name, registry_name, resource_group_name)
2024-10-22T02:44:50.2895388Z /opt/az/lib/python3.11/site-packages/azure/cli/command_modules/acr/_agentpool_polling.py:28: in delete_agentpool_with_polling
2024-10-22T02:44:50.2895772Z     initial_response=client.get(
2024-10-22T02:44:50.2896255Z /opt/az/lib/python3.11/site-packages/azure/core/tracing/decorator.py:94: in wrapper_use_tracer
2024-10-22T02:44:50.2896603Z     return func(*args, **kwargs)
2024-10-22T02:44:50.2897163Z /opt/az/lib/python3.11/site-packages/azure/mgmt/containerregistry/v2019_06_01_preview/operations/_agent_pools_operations.py:336: in get
2024-10-22T02:44:50.2897635Z     map_error(status_code=response.status_code, response=response, error_map=error_map)
2024-10-22T02:44:50.2898016Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2024-10-22T02:44:50.2898157Z 
2024-10-22T02:44:50.2898522Z status_code = 404
2024-10-22T02:44:50.2899107Z response = <RequestsTransportResponse: 404 Not Found, Content-Type: application/json; charset=utf-8>
2024-10-22T02:44:50.2899980Z error_map = {304: <class 'azure.core.exceptions.ResourceNotModifiedError'>, 401: <class 'azure.core.exceptions.ClientAuthenticatio..., 404: <class 'azure.core.exceptions.ResourceNotFoundError'>, 409: <class 'azure.core.exceptions.ResourceExistsError'>}
2024-10-22T02:44:50.2900340Z 
2024-10-22T02:44:50.2900572Z     def map_error(
2024-10-22T02:44:50.2900915Z         status_code: int, response: _HttpResponseCommonAPI, error_map: Mapping[int, Type[HttpResponseError]]
2024-10-22T02:44:50.2901303Z     ) -> None:
2024-10-22T02:44:50.2901546Z         if not error_map:
2024-10-22T02:44:50.2901787Z             return
2024-10-22T02:44:50.2902054Z         error_type = error_map.get(status_code)
2024-10-22T02:44:50.2902330Z         if not error_type:
2024-10-22T02:44:50.2902574Z             return
2024-10-22T02:44:50.2902851Z         error = error_type(response=response)
2024-10-22T02:44:50.2903124Z >       raise error
2024-10-22T02:44:50.2903726Z E       azure.core.exceptions.ResourceNotFoundError: (SubscriptionNotFound) The subscription '00000000-0000-0000-0000-000000000000' could not be found.
2024-10-22T02:44:50.2904144Z E       Code: SubscriptionNotFound
2024-10-22T02:44:50.2904637Z E       Message: The subscription '00000000-0000-0000-0000-000000000000' could not be found.
2024-10-22T02:44:50.2904821Z 
2024-10-22T02:44:50.2905270Z /opt/az/lib/python3.11/site-packages/azure/core/exceptions.py:161: ResourceNotFoundError
```

This is similar to https://github.com/Azure/azure-cli/pull/17185.